### PR TITLE
Problem: not ready for marketplace guideline #82

### PR DESCRIPTION
**Pull request by leejw51crypto**
move to top folder

remove file

folder tidy up

add lfs

add dylib

add progress

add whitelist

fix windows build

add windows dll

### CODE REVIEW

**Review by tomtau — COMMENTED** [review]
should there be those cxx binaries instead of being compiled or downloaded+checksummed from the original repo?

**Comment by damoncro** [comment]
Better to use the tagged release after https://github.com/cronos-labs/play-cpp-sdk/pull/156 is merged.

**Comment by leejw51crypto** [comment]
i'm trying to remove binary in this repo.
and download tagged version in ci build action from play-cpp-sdk


user can download from release , use in ureal

**Comment by damoncro** [comment]
Just tagged v0.0.2-alpha, maybe we can use that after the artifacts are generated.

**Comment by leejw51crypto** [comment]
ok, i'll use v0.0.2-alpha binaries

**Comment by leejw51crypto** [comment]
tagged as "v0.0.2-alpha"

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	mkdir -p ./ThirdParty/cronosplay/lib/Mac
+	mkdir -p ./ThirdParty/cronosplay/lib/Win64
+
+download:
+	cd install/mac && wget $(MACSRC)
+	cd install/mac && tar xvf play_cpp_sdk_Darwin_x86_64.tar.gz
+	cd install/windows && wget $(WINDOWSSRC)
+	cd install/windows && unzip play_cpp_sdk_Windows_x86_64.zip
+	cp install/mac/sdk/lib/* ./ThirdParty/cronosplay/lib/Mac
+	cp install/windows/sdk/lib/* ./ThirdParty/cronosplay/lib/Win64
```
Comment (tomtau) [inline Makefile:19]: + verify checksums?

Comment (leejw51crypto) [reply Makefile]: added sha256sum verification

File: Source/CronosPlayUnreal/CronosPlayUnreal.Build.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      PublicAdditionalLibraries.Add(Path.Combine(
+          LibrariesPath, "cronosplay", "lib", "Win64", "play_cpp_sdk.lib"));
+
+      PublicAdditionalLibraries.Add("Ncrypt.lib");
+      PublicAdditionalLibraries.Add("userenv.lib");
+      PublicAdditionalLibraries.Add("bcrypt.lib");
+      PublicAdditionalLibraries.Add("ntdll.lib");
+      PublicAdditionalLibraries.Add("Secur32.lib");
+      PublicAdditionalLibraries.Add("crypt32.lib");
+      PublicAdditionalLibraries.Add("ws2_32.lib");
```
Comment (tomtau) [inline Source/CronosPlayUnreal/CronosPlayUnreal.Build.cs:48]: these exists on the end platform?

File: Source/CronosPlayUnreal/Private/PlayCppSdkDownloader.cpp [hunk 3]
```diff
@@ -0,0 +1,206 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
Comment (tomtau) [inline Source/CronosPlayUnreal/Private/PlayCppSdkDownloader.cpp:1]: add a copyright or remove?

File: Source/CronosPlayUnreal/Public/PlayCppSdkDownloader.h [hunk 4]
```diff
@@ -0,0 +1,159 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
Comment (tomtau) [inline Source/CronosPlayUnreal/Public/PlayCppSdkDownloader.h:1]: ditto

File: ThirdParty/cronosplay/include/defi-wallet-core-cpp/src/contract.rs.cc [hunk 5]
```diff
@@ -1,6 +1,3 @@
-#pragma warning(disable : 4583)
-#pragma warning(disable : 4582)
```
Comment (tomtau) [inline ThirdParty/cronosplay/include/defi-wallet-core-cpp/src/contract.rs.cc:2]: no need for these pragmas anymore?

Comment (tomtau) [reply ThirdParty/cronosplay/include/defi-wallet-core-cpp/src/contract.rs.cc:2]: plus in other generated code

**Review by tomtau — APPROVED** [review]
it seems some of the generated code removed those pragmas which disabled turning some warnings into errors (due to CXX generated code)... does it still compile ok?
makefile download+checksum verification may be ok -- I was wondering whether that's fine for the marketplace, or should that be inside the unreal build script (or the stuff to be distributed would be the same as what's uploaded by the GH action flow, so it should contain the correct binaries)?

File: Source/CronosPlayUnreal/CronosPlayUnreal.Build.cs (hunk 2 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/CronosPlayUnreal.Build.cs:48]: yes, it's default library from visual studio

File: Source/CronosPlayUnreal/Private/PlayCppSdkDownloader.cpp (hunk 3 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/PlayCppSdkDownloader.cpp:1]: i'll add copyright

File: Source/CronosPlayUnreal/Public/PlayCppSdkDownloader.h (hunk 4 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Public/PlayCppSdkDownloader.h:1]: will add too

File: ThirdParty/cronosplay/include/defi-wallet-core-cpp/src/contract.rs.cc (hunk 5 above)
Comment (leejw51crypto) [reply ThirdParty/cronosplay/include/defi-wallet-core-cpp/src/contract.rs.cc:2]: because we're using static library, not used now

**Pull request merged**